src/EFCore.PG/Storage/Internal/Mapping/NpgsqlImmutableHstoreTypeMapping.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal.Mapping
+{
+    /// <summary>
+    /// The type mapping for the PostgreSQL hstore type to immutable .NET Dictionaries.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/hstore.html
+    /// </remarks>
+    public class NpgsqlImmutableHstoreTypeMapping : NpgsqlHstoreTypeMapping
```

Comment:
Naming-wise this shouldn't be called Immutable (since it doesn't do ImmutableDictionary) - ReadOnly would be better.

Reply:
This is good naming now that we only support ImmutableDictionary, and not IReadOnlyDictionary.